sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -20,25 +24,49 @@
  */
 public class RetryPolicy implements HttpPipelinePolicy {
 
-    private static final String RETRY_AFTER_MS_HEADER = "retry-after-ms";
-
     private final ClientLogger logger = new ClientLogger(RetryPolicy.class);
     private final RetryStrategy retryStrategy;
 
+    private final HttpHeaderName retryAfterHeader;
+
     /**
      * Creates a default {@link ExponentialBackoff} retry policy.
      */
     public RetryPolicy() {
-        this(new ExponentialBackoff());
+        this(new ExponentialBackoff(), HttpHeaderName.AZURE_RETRY_AFTER_MS_HEADER);
+    }
+
+    /**
+     * Creates a default {@link ExponentialBackoff} retry policy.
+     * @param retryAfterHeader The retry after http header name to be used get retry after value from
+     * {@link HttpResponse}.
+     * @throws NullPointerException if {@code retryAfterHeader} is {@code null}.
+     */
+    public RetryPolicy(HttpHeaderName retryAfterHeader) {
+        this(new ExponentialBackoff(), retryAfterHeader);
     }
 
     /**
      * Creates a RetryPolicy with the provided {@link RetryStrategy}.
      *
      * @param retryStrategy The {@link RetryStrategy} used for retries.
+     * @throws NullPointerException if {@code retryStrategy} is {@code null}.
      */
     public RetryPolicy(RetryStrategy retryStrategy) {
+        this(retryStrategy, HttpHeaderName.AZURE_RETRY_AFTER_MS_HEADER);
+    }
+
+    /**
+     * Creates a {@link RetryPolicy} with the provided {@link RetryStrategy} and {@link HttpHeaderName}.
+     *
+     * @param retryStrategy The {@link RetryStrategy} used for retries.
+     * @param retryAfterHeader The retry after http header name to be used get retry after value from
+     * {@link HttpResponse}.
+     * @throws NullPointerException if {@code retryStrategy} or {@code retryAfterHeader} is {@code null}.
+     */
+    public RetryPolicy(RetryStrategy retryStrategy, HttpHeaderName retryAfterHeader) {
         this.retryStrategy = Objects.requireNonNull(retryStrategy, "'retryStrategy' cannot be null");
+        this.retryAfterHeader = Objects.requireNonNull(retryAfterHeader, "'retryAfterHeader' cannot be null");
```

Comment:
This can be `null`. If the header is not provided, then the retry strategy will be used to determine the delay. If provided, the delay will be determined based on the header value. Also, you will need the TimeUnit as well. Can't assume the unit is always milliseconds. For e.g. the default HTTP header `Retry-After` returns delay in seconds.